forte/processors/stave/stave_processor.py

```diff
@@ -0,0 +1,373 @@
+                json={
+                    "name": self.configs.user_name,
+                    "password": self.configs.user_password
+                })
+            logger.info("%d %s", response.status_code, response.text)
+            if response.status_code != 200:
+                return
+
+            # Configure and create project
+            if self._project_id < 0:
+                response = session.post(f"{self._url}/api/projects/new",
+                    json={
+                        "type": self.configs.projectType,
+                        "name": self.configs.projectName,
+                        "ontology": json.dumps(
+                            self.resources.get("onto_specs_dict")
+                        ),
+                        "multiOntology": str(self.configs.multiOntology),
+                        "config": str(self.configs.projectConfigs)
+                    })
+                logger.info("%d %s", response.status_code, response.text)
+                if response.status_code != 200:
+                    return
+                self._project_id = response.json()["id"]
+                webbrowser.open(self._url)
+
+            # Configure and create document
+            response = session.post(f"{self._url}/api/documents/new",
+                json={
+                    "name": input_pack.pack_name,
+                    "textPack": input_pack.serialize(),
+                    "project_id": self._project_id,
+                })
+            logger.info("%d %s", response.status_code, response.text)
+            if response.status_code != 200:
+                return
+
+    @classmethod
+    def default_project_configs(cls, ontology: Dict[str, Any]):
+        """
+        Create default project configuration based on ontology.
+        This is translated from JavaScript function `createDefaultConfig` in
+        https://github.com/asyml/stave/blob/d82383de3d74bf09c0d30f33d8a902595
+        f5aff80/src/app/pages/Projects.tsx#L140
+        Commit link:
+        https://github.com/asyml/stave/commit/63ae90583016bbc7d88c091a3bd8a2b
+        56675e49e#diff-46d2dc8ce1bf31f277e7a217416c0be739924160096aac4ca1fee4
+        23bee4c014R105
+
+        Args:
+            ontology: A dictionary representing ontology.
+
+        Returns:
+            configs: A dictionary with the default config for project.
+
+        """
+        configs: Dict[str, Any] = {
+            "legendConfigs": {},
+            "scopeConfigs": {},
+            "layoutConfigs": {
+                "center-middle": "default-nlp",
+                "left": "default-meta",
+                "right": "default-attribute",
+                "center-bottom": "disable"
+            }
+        }
+
+        legend_configs: Dict[str, Any] = {}
+        entry_graph = collections.defaultdict(set)
+        annotation_entry = "forte.data.ontology.top.Annotation"
+        for entry in ontology["definitions"]:
```

Comment:
I think we already have a tree here: https://github.com/asyml/forte/blob/master/forte/pipeline.py#L263, I think we can reuse this.